WordPress/Classes/ViewRelated/Me/App Settings/AppSettingsViewController.swift

```diff
@@ -220,6 +248,32 @@ class AppSettingsViewController: UITableViewController {
         }
     }
 
+    func imageOptimizationChanged() -> (Bool) -> Void {
+        return { [weak self] value in
+            MediaSettings().imageOptimizationEnabled = value
+
+            // Track setting changes
+            WPAnalytics.track(.appSettingsOptimizeImagesChanged, properties: ["enabled": value])
+
+            // Show/hide image optimization settings
+            guard let self = self, let tableView = self.tableView else {
+                return
+            }
+            tableView.beginUpdates()
```

Comment:
It looks like we can make this smoother if we set `handler.automaticallyReloadTableView = false` - maybe right after [it's set](https://github.com/wordpress-mobile/WordPress-iOS/blob/63d625037193b46fbaecc61701579d58b7f8076f/WordPress/Classes/ViewRelated/Me/App%20Settings/AppSettingsViewController.swift#L48). I think the call to `reloadData()` [in the setter](https://github.com/wordpress-mobile/WordPress-iOS/blob/63d625037193b46fbaecc61701579d58b7f8076f/WordPress/Classes/Utility/ImmuTable.swift#L260) is causing trouble.

After setting `automaticallyReloadTableView` to `false` as noted above, and doing the following:

```swift
            tableView.performBatchUpdates {
                self.reloadViewModel()
                let imageSizingIndex = IndexPath(row: 1, section: 0)
                let imageQualityIndex = IndexPath(row: 2, section: 0)
                let imageOptimizationSettingsIndexes = [imageSizingIndex, imageQualityIndex]
                tableView.reloadSections(IndexSet(integer: 0), with: .automatic)
            }
```

The animations worked as expected:

https://github.com/wordpress-mobile/WordPress-iOS/assets/2092798/f286bf50-64ef-4a3c-b943-6d9e69100c4e

⚠️ However, I don't know how disabling automatically reloading will affect existing functionality in the view. If we spot any issues we could just remove animations for now and address them later.

Reply:
> It looks like we can make this smoother if we set `handler.automaticallyReloadTableView = false` - maybe right after [it's set](https://github.com/wordpress-mobile/WordPress-iOS/blob/63d625037193b46fbaecc61701579d58b7f8076f/WordPress/Classes/ViewRelated/Me/App%20Settings/AppSettingsViewController.swift#L48). I think the call to `reloadData()` [in the setter](https://github.com/wordpress-mobile/WordPress-iOS/blob/63d625037193b46fbaecc61701579d58b7f8076f/WordPress/Classes/Utility/ImmuTable.swift#L260) is causing trouble.

I tried to set `automaticallyReloadTableView` to `false` after [it's set](https://github.com/wordpress-mobile/WordPress-iOS/blob/63d625037193b46fbaecc61701579d58b7f8076f/WordPress/Classes/ViewRelated/Me/App%20Settings/AppSettingsViewController.swift#L48) but leads to omit updating values in the table after changing them.

**Example:**

https://github.com/wordpress-mobile/WordPress-iOS/assets/14905380/e704cbb3-0b4f-4b5a-be4d-ac39b25dd9cf

_Note that _Image Quality_ and _Max Video Upload Size_ settings don't reflect the value selected._

The approach I found that works is to disable `automaticallyReloadTableView` before reloading the view model and set it to the original value after that. Changes applied in https://github.com/wordpress-mobile/WordPress-iOS/pull/21981/commits/132bfbcc2da4ec94a36a7caa4a6e285de06aa484.

https://github.com/wordpress-mobile/WordPress-iOS/assets/14905380/6f5735d5-4faa-43c1-ab0f-2ad8debf67c6

Reply:
Good spot @fluiddot! Works for me. 👍